client/network/test/src/sync.rs

```diff
@@ -814,3 +811,32 @@ fn sync_to_tip_requires_that_sync_protocol_is_informed_about_best_block() {
 	// However peer 1 should still not have the block.
 	assert!(!net.peer(1).has_block(&block_hash));
 }
+
+#[test]
+fn make_sure_we_sync_to_tip_without_informing_sync_about_new_best_block() {
+	sp_tracing::try_init_simple();
+
+	let mut net = TestNet::new(3);
+
+	let block_hash = net.peer(0).push_blocks_at_without_informing_sync(
+		BlockId::Number(0),
+		10_000,
+		false,
+	);
+
+	net.peer(1).push_blocks_at_without_informing_sync(
+		BlockId::Number(0),
+		5_000,
+		false,
+	);
+
+	net.block_until_connected();
+	net.block_until_idle();
+
+	assert!(!net.peer(2).has_block(&block_hash));
+
+	net.peer(0).network_service().new_best_block_imported(block_hash, 10_000);
```

Comment:
Aren't we informing sync about new best block here?

Reply:
I think it's intentional. You *have* to inform the sync at some point for the thing to work.

Reply:
My comment was because the test was named `make_sure_we_sync_to_tip_without_informing_sync_about_new_best_block`, it's been renamed since :)